crawl/workspace.py

```diff
@@ -170,19 +160,25 @@ def _parse_maven_install(self, maven_install_info, repo_root_path):
                 transitives_list.append({dep : transitives})
                 self.dependency_metadata.register_exclusions(dep, exclusions)
 
-        # Overrides the deps honoring the override file
-        for key, dep in result.items():
+        # Overrides the direct deps mapping
+        overridden_result = copy.deepcopy(result)
+
+        for key, dep in overridden_result.items():
             if self.override_file_info == []:
                 break
             overridden_dep = self.override_file_info.overridden_dep_value(dep)
-            if overridden_dep in result.keys():
-                result[key] = result[overridden_dep]
+            if overridden_dep in overridden_result.keys():
+                overridden_result[key] = overridden_result[overridden_dep]
 
         # Registers the overridden transitives
         for t in transitives_list:
             for dep, transitives in t.items():
                 if not self.override_file_info == []:
-                    transitives = self.override_file_info.override_deps(transitives, result)
+                    transitives = self.override_file_info.override_deps(transitives, overridden_result)
                 self.dependency_metadata.register_transitives(dep, transitives)
 
-        return result
\ No newline at end of file
+        # Registers direct deps with their overridden deps
+        # If a dep is not be overridden then it will point to the original dep
+        self.label_to_ext_deps = overridden_result
```

Comment:
It is confusing to set class state and to also return a value, I think.  The method could either not return any values and only set class state, or return multiple values and not mess with class state.  I prefer the latter: let's return 2 values and assign the state on the caller side.